test/Microsoft.AspNetCore.JsonPatch.Test/NestedObjectTests.cs

```diff
@@ -1802,6 +1947,71 @@ public void MoveInListWithSerialization()
         }
 
         [Fact]
+        public void MoveReferenceInList()
+        {
+            // Arrange
+            var sDto1 = new SimpleDTO() { IntegerValue = 1 };
+            var sDto2 = new SimpleDTO() { IntegerValue = 2 };
+            var sDto3 = new SimpleDTO() { IntegerValue = 3 };
+            var doc = new SimpleDTOWithNestedDTO()
+            {
+                SimpleDTOList = new List<SimpleDTO>() {
+                    sDto1,
+                    sDto2,
+                    sDto3
+                }
+            };
+
+            // create patch
+            var patchDoc = new JsonPatchDocument<SimpleDTOWithNestedDTO>();
+            patchDoc.Move<SimpleDTO>(o => o.SimpleDTOList, 0, o => o.SimpleDTOList, 1);
+
+            // Act
+            patchDoc.ApplyTo(doc);
+
+            // Assert
+            Assert.Equal(new List<SimpleDTO>() { sDto2, sDto1, sDto3 }, doc.SimpleDTOList);
+            Assert.Equal(2, doc.SimpleDTOList[0].IntegerValue);
+            Assert.Equal(1, doc.SimpleDTOList[1].IntegerValue);
+            Assert.Same(sDto2, doc.SimpleDTOList[0]);
+            Assert.Same(sDto1, doc.SimpleDTOList[1]);
+        }
+
+        [Fact]
+        public void MoveReferenceInListWithSerialization()
+        {
+            // Arrange
+            var sDto1 = new SimpleDTO() { IntegerValue = 1 };
+            var sDto2 = new SimpleDTO() { IntegerValue = 2 };
+            var sDto3 = new SimpleDTO() { IntegerValue = 3 };
+            var doc = new SimpleDTOWithNestedDTO()
+            {
+                SimpleDTOList = new List<SimpleDTO>() {
+                    sDto1,
+                    sDto2,
+                    sDto3
+                }
+            };
+
+            // create patch
+            var patchDoc = new JsonPatchDocument<SimpleDTOWithNestedDTO>();
+            patchDoc.Move<SimpleDTO>(o => o.SimpleDTOList, 0, o => o.SimpleDTOList, 1);
+
+            var serialized = JsonConvert.SerializeObject(patchDoc);
+            var deserialized = JsonConvert.DeserializeObject<JsonPatchDocument<SimpleDTOWithNestedDTO>>(serialized);
+
+            // Act
+            patchDoc.ApplyTo(doc);
+
+            // Assert
+            Assert.Equal(new List<SimpleDTO>() { sDto2, sDto1, sDto3 }, doc.SimpleDTOList);
+            Assert.Equal(2, doc.SimpleDTOList[0].IntegerValue);
+            Assert.Equal(1, doc.SimpleDTOList[1].IntegerValue);
+            Assert.Same(sDto2, doc.SimpleDTOList[0]);
+            Assert.Same(sDto1, doc.SimpleDTOList[1]);
+        }
+
+        [Fact]
```

Comment:
We would need tests for the other adapters which are impacted by the changes here: Poco and Expando